pentasharp/Services/BusReservationService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using pentasharp.Data;
+using pentasharp.Interfaces;
+using pentasharp.Models.Enums;
+using pentasharp.ViewModel.BusReservation;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace pentasharp.Services
+{
+    public class BusReservationService : IBusReservationService
+    {
+        private readonly AppDbContext _context;
+
+        public BusReservationService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<object>> GetReservationsAsync(int companyId)
+        {
+            var reservations = await _context.BusReservations
+                .Where(r => r.Schedule != null && r.Schedule.Route != null && r.Schedule.Bus != null && r.BusCompanyId == companyId)
+                .Select(r => new
+                {
+                    ReservationId = r.ReservationId,
```

Comment:
i think r.ReservationId is directly mapped to ReservationId without explicitly stating the assignment